### DISCUSSION TITLE
The way not to start driver until self.driver.uc_open_with_reconnect()

### DISCUSSION THREAD

**Discussion by LiPingYen** [seleniumbase/SeleniumBase#2605] [Q&A] [answered]
When I run the code with `pytest`, the driver is initiated before `self.driver.uc_open_with_reconnect()`.  I only want to open the driver when the conditions are met so I add `if else` in my code. Is there any possible way not to open the driver until `self.driver.uc_open_with_reconnect()` is called?

**Comment by mdmintz** [comment, marked as answer]
The `pytest` formats all spin up the browser at the beginning of the test (and close it at the end).

If you want to spin up the browser at specific times, use the `SB(uc=True)` or `Driver(uc=True)` formats.
More info in [help_docs/uc_mode.md](https://github.com/seleniumbase/SeleniumBase/blob/master/help_docs/uc_mode.md) and at the end of [help_docs/syntax_formats.md](https://github.com/seleniumbase/SeleniumBase/blob/master/help_docs/syntax_formats.md)
